tsflex/features/feature_collection.py

```diff
@@ -260,14 +268,98 @@ def add(
-        # global get_stroll_func
+    def _executor_stroll(idx: int) -> pd.DataFrame:
+        """Executor function for the StridedRolling.apply_func method.
+
+        Strided rolling feature calculation occurs when either;
+        - a `window` and `stride` argument is stored in the `FeatureDescriptor` object
+          (or the `stride` argument is passed to the `calculate` method)
+        - segment indices are passed to the `calculate` method
+        - a `group_by_consecutive` argument is passed to the `calculate` method (since
+          we calculate the segment indices for the consecutive groups)
+        """
+        # Uses the global get_stroll_func
         stroll, function = get_stroll_func(idx)
-        return stroll.apply_func(function)
+        return stroll.apply_func(function)  # execution time is logged in apply_func
+
+    @staticmethod
+    def _executor_grouped(idx: int) -> pd.DataFrame:
+        """Executor function for grouped feature calculation.
+
+        Grouped feature calculation occurs when either;
+        - a `group_by_all` argument is passed to the `calculate` method
+        - a `DataFrameGroupBy` is passed as `data` argument to the `calculate` method
+
+        Note that passing a `group_by_consecutive` argument to the `calculate` method
+        will not use this executor function, but will use the `_executor_stroll` as
+        executor function (since we calculate the segment indices for the consecutive
+        groups).
+        """
+        # Uses the global get_group_func, group_indices, and group_id_name
+        data, function = get_group_func(idx)
+        group_ids = group_indices.keys()  # group_ids are the keys of the group_indices
+        cols = data.columns.values
+
+        t_start = time.perf_counter()
+
+        # Wrap the function to handle multiple inputs and / or convert to numpy array
+        # if necessary
+        f = function
+        if len(cols) == 1 and function.input_type is np.array:
+
+            def f(x: pd.DataFrame):
+                return function(x.values)
+
+        elif len(cols) > 1:
+            if function.input_type is np.array:
+
+                def f(x: pd.DataFrame):
+                    return function(*[x[c].values for c in cols])
+
+            else:  # function.input_type is pd.Series
+
+                def f(x: pd.DataFrame):
+                    return function(*[x[c] for c in cols])
+
+        # Function execution over the grouped data (accessed by using the group_indices)
+        out = np.array(list(map(f, [data.iloc[idx] for idx in group_indices.values()])))
+
+        # Aggregate function output in a dictionary
+        output_names = ["|".join(cols) + "__" + o for o in function.output_names]
+        feat_out = _process_func_output(out, group_ids, output_names, str(function))
+        # Log the function execution time
+        _log_func_execution(
+            t_start, function, tuple(cols), "groupby_all", "groupby_all", output_names
```

Comment:
```suggestion
            t_start, function, tuple(cols), "manual", "manual", output_names
```

This complies more with the `group_by_consecutive` logging functionality (which is internally the same logging functionality as for the custom segment indices)